django/db/models/fields/related_descriptors.py

```diff
@@ -1280,8 +1281,15 @@ def add(self, *objs, through_defaults=None):
                         self.source_field_name,
                         *objs,
                         through_defaults=through_defaults,
+                        using=db,
+                        raw=raw,
                     )
 
+        def add(self, *objs, through_defaults=None):
+            self._remove_prefetched_objects()
+            db = router.db_for_write(self.through, instance=self.instance)
```

Comment:
Do we need to check this here as well as at the top of `_add_base()` (or just once)?